### PR TITLE
feat: archiver 3

### PR DESCRIPTION
**Pull request by hacdias**
- Remove dependency on go hugo by merging front-matter editor with content.
- Update to use archiver 3.

Ref.: #554 https://github.com/filebrowser/frontend/pull/39

### CODE REVIEW

**Comment by hacdias** [comment]
[WIP] not working yet

**Comment by eine** [comment]
I'm not really sure about what's going on here. Let me word what I understood:

- The API of Hugo changed and that broke the build of filebrowser, because we are not using any vendoring/module system ATM.
- At the same time, the archiver dependency also changed, so the build also broke.
- Some users have worked around it by cloning specific versions of hugo and archiver.

If that is ok, the most straigthforward solution would be #496. By using go modules, the build would work without further modifications.

However, at the same time @hacdias suggested getting rid of hugo as a dependency. I think that it makes sense, because AFAIK it is only used for a slightly improved editing experience. I.e., should the site be served, hugo is required as an external dependency anyway.

I am not sure about the relation between archiver and hugo. @hacdias, should these commits be split to separate PRs? Moreover, in the last comment you said that it is not working, yet. Is it because of the frontmatter/hugo part or because of the archiver?

**Comment by hacdias** [comment]
@1138-4EB not working because of archiver. The resultant archives are empty and I still don't know why. Yes, it would be better to separate this into two different PRs. Also, using a vendoring system ASAP would be good.

Does Caddy build server support go.mod @mholt?

File: lib/http/download.go [hunk 1]
```diff
@@ -45,21 +45,21 @@ func downloadHandler(c *fb.Context, w http.ResponseWriter, r *http.Request) (int
...
 		extension string
-		ar        archiver.Archiver
+		ar        archiver.Writer
 	)
 
 	switch query {
 	// If the format is true, just set it to "zip".
 	case "zip", "true", "":
-		extension, ar = ".zip", archiver.Zip
+		extension, ar = ".zip", &archiver.Zip{}
```
Comment (mholt) [inline lib/http/download.go:54]: I recommend using `archiver.DefaultZip` (or if you use more than one, probably `archiver.NewZip()`) -- it sets some sane defaults. Probably especially important for TarGz, which has a default compression level of 0 without sane defaults :wink:

**Comment by mholt** [comment]
> Does Caddy build server support go.mod @mholt?

No, but you can technically vendor some dependencies you need... just make sure they don't conflict with any others that are vendored...

File: lib/http/download.go (hunk 1 above)
Comment (hacdias) [reply lib/http/download.go:54]: Thank you @mholt 😄

**Comment by hacdias** [comment]
Closing in favour of #575

**Pull request closed without merging**